samlang-core/src/printer/printer-js.ts

```diff
@@ -21,9 +28,30 @@ export const highIRStatementToString = (highIRStatement: HighIRStatement): strin
     }
     case 'HighIRFunctionCallStatement': {
       const { functionArguments, functionExpression, returnCollector } = highIRStatement;
-      return `var ${returnCollector} = ${highIRExpressionToString(
-        functionExpression
-      )}(${functionArguments.map((arg) => highIRExpressionToString(arg)).join(', ')});`;
+      let functionName = highIRExpressionToString(functionExpression);
+      switch (functionName) {
+        case ENCODED_FUNCTION_NAME_INT_TO_STRING: {
+          functionName = 'String';
+          break;
+        }
+        case ENCODED_FUNCTION_NAME_PRINTLN: {
+          functionName = 'console.log';
+          break;
+        }
+        case ENCODED_FUNCTION_NAME_STRING_TO_INT: {
+          functionName = 'parseInt';
```

Comment:
I think we need to wrap the result with `BigInt`. Also on a related problem, we should translate int literal with something like `BigInt("92131231312312313123")`, otherwise there might be loss of precision problems.

Therefore, it's probably easier to keep the function name here, but inject code in the beginning like

```js
const _builtin_stringConcat = (a, b) => a + b;
const _builtin_println = (v) => console.log(v);
```

This strategy is especially important for println function, because we might want to an alternative implementation for testing. In that version, println should keep the printed stuff in a global variable, so that it can be later read by the test suite.